### PR TITLE
Add ADDRESS_BOOK feature

### PR DESCRIPTION
**Pull request by habibitcoin**

### CODE REVIEW

File: notifications/telegram.js [hunk 1]
```diff
@@ -31,6 +35,33 @@ async function initCommands() {
         await telegramBot.sendMessage(chatId, payment_details, { parse_mode: 'HTML' })
         await telegramBot.sendMessage(chatId, payment_address)
     });
+    telegramBot.onText(/\/withdraw (.+)/, async (msg, match) => {
```
Comment (topether21) [inline notifications/telegram.js:38]: Small suggestion

- Print the error message and show instructions on an empty command (**/\/withdraw ?(.*)/**).

Captura de pantalla 2024-01-09 a la(s) 4 16 13 p  m

```
    telegramBot.onText(/\/withdraw ?(.*)/, async (msg, match) => {
        const chatId = msg.chat.id;

        // Extracting name and amount from the command
        const args = match[1].split(" ");
        if (args.length < 2) {
            await telegramBot.sendMessage(chatId, "Usage: /withdraw [name] [amount]");
            return;
        }

        const name = args[0];
        const amount = parseFloat(args[1]);

        if (isNaN(amount) || amount <= 1) {
            await telegramBot.sendMessage(chatId, "Please enter a valid amount in satoshis.");
            return;
        }

        try {
            // Call your create_withdraw_request function
            const { withdrawal_details } = await create_withdraw_request(name, amount);
            await telegramBot.sendMessage(chatId, withdrawal_details, { parse_mode: "HTML" });
        } catch (error) {
            console.error("Error processing withdrawal request:", error);
            await telegramBot.sendMessage(chatId, `An error occurred while processing your request. ${error.message}.`);
        }
    });
```

**Review by topether21 — APPROVED** [review]
Awesome work!

File: notifications/telegram.js (hunk 1 above)
Comment (habibitcoin) [reply notifications/telegram.js:38]: great idea, thx! added

**Comment by dannydeezy** [comment]
nice but suggest denominating the telegram command in BTC

**Pull request merged**